### ISSUE TITLE
Multi-line cursor support

### ISSUE THREAD

**Issue by PThorpe92** [sayanarijit/tui-input#21]
Hey first just want to say thanks for this! :D I'm currently using this (crossterm) + ratatui to build a libcurl HTTP client, and it has served my purposes quite well so far.

In order to support adding larger JSON request bodies, I find myself in need of multi-line input/cursor support, and I was digging around in the crossterm docs and saw that they have `cursor::MoveToNextLine` and `cursor::MoveToPreviousLine` available in their API. 
After some playing around I realize that unless I'm really missing something, implementing this as it stands would require a decent amount of change, obviouisly adding a `Y` coordinate to the `cursor` field of the main `input` struct.

I am currently having a go at implementing this, I was just curious if I get it working this might be a PR/change that you might be interested in?

One thing I will say is that I just now started really digging into this and it's also quite possible that I am missing a workaround or something similar

**Comment by sayanarijit** [comment]
Hi. Multiline cursor is out of scope for this project. For simple use cases, you can combine multiple input instances to implement basic multiline functionalities. There's also https://github.com/rhysd/tui-textarea which might interest you.

**Comment by sayanarijit** [comment]
That said, if the above options aren't solving your usecases, I may not be getting the requirements right. In that case, please feel free to suggest improvements via code snippets or pull request including a demo of the problem to be solved.

**Comment by PThorpe92** [comment]
Thanks, yeah I looked into that other library as well but a combination of lack of support for text wrapping and the fact that I already had this as a dependency I figured I'd see about making it work. 

I'll probably either end up hosting a fork or just writing my own crossterm specific simple multi-line text editor. If I use this as a starting point and get this working properly I'll submit a PR and just see if it's something you are interested in. It's obviously going to be quite a drastic change to the current API. 

Either way thanks for the reply :) I'll close this and get in touch if things work out.
